File: 11. Container With Most Water/solution.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
  let max = 0;
  let left = 0;
  let right = height.length - 1;

  while (right > left) {
    max = Math.max(max, (right - left) * Math.min(height[left], height[right]));
    if (height[left] < height[right]) left++;
    else right--;
  }

  return max;
};

File: 11. Container With Most Water/solution.go
func maxArea(height []int) int {
	max := 0
	left, right := 0, len(height)-1

	for right > left {
		temp := 0
		if dist := right - left; height[left] > height[right] {
			temp = dist * height[right]
			right--
		} else {
			temp = dist * height[left]
			left++
		}

		if temp > max {
			max = temp
		}
	}

	return max
}